qiita_pet/util.py

```diff
@@ -51,8 +56,13 @@ def clean_str(item):
     -------
     str
         cleaned string
+
+    Notes
+    -----
+    This function removes colons and periods from strings, and replaces spaces
+    with underscores.
     """
-    return str(item).replace(" ", "_").replace(":", "")
+    return re.sub(clean_str_regex, "", str(item).replace(" ", "_"))
```

Comment:
oh, that sucks :(

what about a translation table?

On Tue, Feb 24, 2015 at 1:00 PM, adamrp <notifications@github.com> wrote:

> In qiita_pet/util.py
> <https://github.com/biocore/qiita/pull/865#discussion_r25286695>:
>
> >      """
> > -    return str(item).replace(" ", "_").replace(":", "")
> > +    return re.sub(clean_str_regex, "", str(item).replace(" ", "_"))
>
> Unfortunately no, since the . and : are being replaced with something
> different than the space (empty string versus underscore, respectively). It
> could be implemented like this:
>
> clean_str_regex = re.compile("[.:]")def repl_fn(matchobj):
>     if matchobj.group(0) == ' ':
>         return '_'
>     else:
>         return ''
> def clean_str(item):
>     return re.sub(clean_str_regex, repl_fn, str(item))
>
> but I'm not sure if that would be faster (it's calling a function for each
> occurrence of any of the three characters specified).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/865/files#r25286695>.
>